src/applications/vaos/appointment-list/components/CanceledAppointmentsList.jsx

```diff
@@ -0,0 +1,153 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import LoadingIndicator from '@department-of-veterans-affairs/component-library/LoadingIndicator';
+import AlertBox from '@department-of-veterans-affairs/component-library/AlertBox';
+import recordEvent from 'platform/monitoring/record-event';
+import * as actions from '../redux/actions';
+import {
+  selectFeatureRequests,
+  selectIsCernerOnlyPatient,
+} from '../../redux/selectors';
+import {
+  selectFutureStatus,
+  selectCanceledAppointments,
+} from '../redux/selectors';
+import {
+  FETCH_STATUS,
+  GA_PREFIX,
+  APPOINTMENT_TYPES,
+} from '../../utils/constants';
+import { getVAAppointmentLocationId } from '../../services/appointment';
+import AppointmentListItem from './AppointmentsPage/AppointmentListItem';
+import ExpressCareListItem from './AppointmentsPage/ExpressCareListItem';
+import NoAppointments from './NoAppointments';
+import moment from 'moment';
+
+function CanceledAppointmentsList({
+  showScheduleButton,
+  isCernerOnlyPatient,
+  appointmentsByMonth,
+  futureStatus,
+  facilityData,
+  fetchFutureAppointments,
+  startNewAppointmentFlow,
+}) {
+  useEffect(
+    () => {
+      if (futureStatus === FETCH_STATUS.notStarted) {
+        fetchFutureAppointments();
+      }
+    },
+    [fetchFutureAppointments, futureStatus],
+  );
+
+  if (futureStatus === FETCH_STATUS.loading) {
```

Comment:
Might be worth including a check for `FETCH_STATUS.notStarted` in here, otherwise we might flash the `NoAppointments` component on first render.